### PR TITLE
Add topologies supporting spatial analysis

### PR DESCRIPTION
**Pull request by naiteeks**

### CODE REVIEW

**Comment by naiteeks** [comment]
@avkewalr-msft , @anilmur  - please review.

File: MediaGraph/topologies/lva-spatial-analysis/readme.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Analyzing Live Video with Computer Vision for Spatial Analysis
+
+The video from an RTSP-capable camera is sent to an external module (spatial analysis) which carries out an AI operation supported by the external module. In the video sample, when the criteria defined by the AI operation is met, events are sent to a signal gate processor which opens, sending the frames to an AMS asset sink. As a result, new Assets are created containing clips where the criteria defined by the AI operation are met.
```
Comment (anilmur) [inline MediaGraph/topologies/lva-spatial-analysis/readme.md:3]: Add link from "spatial analysis" to Cog Svc page. Reword 1st sentence so as to not repeat "external module".

What do you mean by "In the video sample"???

Remove "AMS", it should be "..., sending video frames to an asset sink node. "

Add a line saying use of this topo is shown in a tutorial<link>, where details of "AI operation" are also provided

File: MediaGraph/topologies/lva-spatial-analysis/topology.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{
+    "@apiVersion": "1.0",
+    "name": "InferencingWithCVExtension",
+    "properties": {
+      "description": "Analyzing live video using spatialAnalysis Extension to send images to an external inference engine",
```
Comment (anilmur) [inline MediaGraph/topologies/lva-spatial-analysis/topology.json:5]: typo spatialAnalysis. 

Stay consistent with the topic title, "Analyzing Live Video with Computer Vision for Spatial Analysis"

File: MediaGraph/topologies/lva-spatial-analysis/topology.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        {
+          "name": "grpcUrl",
+          "type": "String",
+          "description": "inferencing Url",
+          "default": "tcp://spatialAnalysis:50051"
+        },
+        {
+          "name": "frameRate",
+          "type": "String",
+          "description": "Rate of the frames per second to be received from LVA.",
```
Comment (anilmur) [inline MediaGraph/topologies/lva-spatial-analysis/topology.json:33]: This is a setting for LVA. So it is the "Rate... to be sent to the Spatial Analysis module"

**Review by anilmur — CHANGES_REQUESTED** [review]

File: MediaGraph/topologies/lva-spatial-analysis/readme.md (hunk 1 above)
Comment (naiteeks) [reply MediaGraph/topologies/lva-spatial-analysis/readme.md]: added the links now.

File: MediaGraph/topologies/lva-spatial-analysis/topology.json (hunk 2 above)
Comment (naiteeks) [reply MediaGraph/topologies/lva-spatial-analysis/topology.json]: updated the description.

File: MediaGraph/topologies/lva-spatial-analysis/topology.json (hunk 3 above)
Comment (naiteeks) [reply MediaGraph/topologies/lva-spatial-analysis/topology.json]: fixed the sentence.

File: MediaGraph/topologies/lva-spatial-analysis/readme.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Analyzing Live Video with Computer Vision for Spatial Analysis
+
+The video from an RTSP-capable camera is sent to an external module ([spatial analysis](https://docs.microsoft.com/en-us/azure/cognitive-services/computer-vision/spatial-analysis-operations)) which carries out an AI operation supported by the external module. From the video, when the criteria defined by the AI operation is met, events are sent to a signal gate processor which opens, sending the frames to an asset sink node. As a result, new Assets are created containing clips where the criteria defined by the AI operation are met. You can read more about how to use Live Video Analytics together with Computer Vision for spatial analysis AI service from Azure Cognitive Services in [this](https://aka.ms/lva-spatial-analysis) documentation page.
```
[the remark this answers is not in the document]
Comment (anilmur) [reply MediaGraph/topologies/lva-spatial-analysis/readme.md:3]: "From the video, when..." does not read right

File: MediaGraph/topologies/lva-spatial-analysis/topology.json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        {
+          "name": "grpcUrl",
+          "type": "String",
+          "description": "inferencing Url",
+          "default": "tcp://spatialAnalysis:50051"
+        },
+        {
+          "name": "frameRate",
+          "type": "String",
+          "description": "Rate of the frames per second to send to the spatialAnalysis module.",
```
Comment (anilmur) [inline MediaGraph/topologies/lva-spatial-analysis/topology.json:33]: Limits the rate at which frames are sent to the Spatial Analysis module

Note: you are not using the term "spatialAnalysis" anywhere else in this topology

**Review by anilmur — CHANGES_REQUESTED** [review]

File: MediaGraph/topologies/lva-spatial-analysis/topology.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        {
+          "name": "grpcUrl",
+          "type": "String",
+          "description": "inferencing Url",
+          "default": "tcp://spatialAnalysis:50051"
+        },
+        {
+          "name": "frameRate",
+          "type": "String",
+          "description": "Limit the rate of the frames per second to send to the computerVisionExtension.",
```
Comment (anilmur) [inline MediaGraph/topologies/lva-spatial-analysis/topology.json:33]: Limit the rate (per second) at which frames are sent to the MediaGraphCognitiveServicesVisionExtension node

**Review by anilmur — COMMENTED** [review]
#sign-off

**Pull request merged**